pallets/file-system/src/utils.rs

```diff
@@ -872,6 +893,38 @@ where
             }
         };
 
+        // Add the pending stop storing request to storage.
+        <PendingStopStoringRequests<T>>::insert(
+            &bsp_id,
+            &file_key,
+            (frame_system::Pallet::<T>::block_number(), size),
+        );
+
+        Ok(bsp_id)
+    }
+
+    pub(crate) fn do_bsp_confirm_stop_storing(
```

Comment:
I thought about that but didn't find an easy way to achieve that.
We can check for the non-existence of the storage request, and that would work great if the storage request was created by the request to stop storing from the BSP (because it would be a storage request with only one BSP required), but if the storage request is a normal one we are basically locking the BSP from stop storing a file until it is completely stored by the system, which could take quite a while